src/Sarif/UriHelper.cs

```diff
@@ -69,6 +71,16 @@ public static string MakeValidUri(string path)
                     !path.StartsWith(@"\", StringComparison.Ordinal))
                 {
                     validUri = validUri.Substring(1);
+
+                    // When the UriBuilder constructs an absolute URI, it strips any
+                    // leading "." and ".." segments ("dot-segments", as RFC 3986 calls
+                    // them). Glue them back on so we don't lose the relative path
+                    // information.
+                    string leadingDotSegments = GetLeadingDotSegments(path);
+                    if (!string.IsNullOrEmpty(leadingDotSegments))
+                    {
+                        validUri = leadingDotSegments + validUri;
+                    }
```

Comment:
Yes, IIRC there was already a bug in the relative path case: that `foo\bar\baz.c` serialized as `/foo/bar/baz.c`. So I added the logic to remove the leading slash -- unless the _original_ path started with a slash.

Again, since you never encountered relative paths, you wouldn't have seen this.

I think syncing the helpers is a good idea, and I think the version I have here is good (and has a good set of tests).